docs/tutorials/build/sync-table.md

```diff
@@ -0,0 +1,982 @@
+
+    You'll define `AuthorSchema` in the next step, but for now just set it as if it already exists.
+
+    </div>
+    <div markdown>
+
+    ```{.ts hl_lines="8-11"}
+    const BookSchema = coda.makeObjectSchema({
+      properties: {
+        // ...
+        subjects: {
+          type: coda.ValueType.Array,
+          items: { type: coda.ValueType.String }
+        },
+        authors: {
+          type: coda.ValueType.Array,
+          items: AuthorSchema,
+        },
+      },
+      // ...
+    });
+    ```
+
+    </div>
+    </section>
+
+=== ":material-numeric-3-circle: Define author schema"
+
+    <section class="tutorial-row" markdown>
+    <div markdown>
+
+    Just like `BookSchema` must be defined before the sync table where it is used, so must `AuthorSchema` be defined before it is used in `BookSchema`.
+
+    Define the properties for each author to match the data returned in the API, using `fromKey` when you want to expose a diffent name to users.
+
+    The schema needs `displayProperty` set to determine which property to show in the chip, but it doesn't need `idProperty` set since it's not being used as a sync table row (just a sub-object inside of row).
+
+    </div>
+    <div markdown>
+
+    ```{.ts hl_lines="1-14"}
+    const AuthorSchema = coda.makeObjectSchema({
+      properties: {
+        name: { type: coda.ValueType.String },
+        born: {
+          type: coda.ValueType.Number,
+          fromKey: "birth_year",
+        },
+        died: {
+          type: coda.ValueType.Number,
+          fromKey: "death_year",
+        },
+      },
+      displayProperty: "name",
+    });
+
+    const BookSchema = coda.makeObjectSchema({
+      // ...
+    });
+    ```
+
+    </div>
+    </section>
+
+=== ":material-numeric-4-circle: Add thumbnail and link"
+
+    <section class="tutorial-row" markdown>
+    <div markdown>
+
+    Next add the thumbnail image for each book, as well as a link to the book on the Project Gutenberg website.
+
+    These fields don't exist in the row data exactly as needed, but you can define "synthetic" properties that you populate manually in the `execute` function (you'll do that in the next step).
+
+    Define a `thumbnail` and `link` property, setting the `codaType` field to instruct Coda to display these as Image URL and Link colunns respectively.
+
+    </div>
+    <div markdown>
+
+    ```{.ts hl_lines="16-23"}
+    const AuthorSchema = coda.makeObjectSchema({
+      // ...
+    });
+
+    const BookSchema = coda.makeObjectSchema({
+      properties: {
+        // ...
+        subjects: {
+          type: coda.ValueType.Array,
+          items: { type: coda.ValueType.String }
+        },
+        authors: {
+          type: coda.ValueType.Array,
+          items: AuthorSchema,
+        },
+        thumbnail: {
+          type: coda.ValueType.String,
+          codaType: coda.ValueHintType.ImageReference,
```

Comment:
Should we recommend ImageAttachment here? We want to discourage hotlinking.

Reply:
Interesting, I had thought that `ImageAttachment` had more issues these days, with the virus scanner being overwhelmed, etc. Changed to `ImageAttachment`.